websocket/handlers/signin.handler.js

```diff
@@ -0,0 +1,39 @@
+import geohash from "ngeohash";
+import {
+  uniqueNamesGenerator,
+  adjectives,
+  colors,
+  animals,
+} from "unique-names-generator";
+
+export const handleSignin = (socket, grids, finalCallback) => (args) => {
+  const { lat, lon } = args || {};
+
+  if (!lat || !lon) {
+    return;
+  }
+
+  const hash = geohash.encode(lat, lon, 5);
+
+  const randomName = uniqueNamesGenerator({
+    dictionaries: [adjectives, colors, animals],
+    separator: "-",
+  }); // big-red-donkey
+
+  let grid = grids[hash];
+
+  if (!grid) {
+    grids[hash] = new Set();
+    grid = grids[hash];
+  }
+
+  grid.add({
+    username: randomName,
+    hash,
+    status: true,
+  });
+
+  socket.emit("signin", { hash, username: randomName });
+
+  finalCallback?.({ hash, username: randomName });
```

Comment:
I'm not sure if this adheres to a specific design pattern, but rather than having these side effects within this function, I decided to separate them from the core logic. I think this approach promotes better code organization and maintainability by isolating the side effects in a modular way. 

Let me know what you think :)

Reply:
The final callback only has two lines that are doing something, `socket.join(hash);` and `io.to(hash).emit("users", [...grids[userHash]]);` as such I think that should just be included here, right now that is adding too much indirection and making this a bit hard to follow and reason about the control flow.